service/awsconfig/v10/resource/s3bucket/resource.go

```diff
@@ -81,17 +81,27 @@ func (r *Resource) Name() string {
 	return Name
 }
 
-func toBucketState(v interface{}) (BucketState, error) {
+func toBucketState(v interface{}) ([]BucketState, error) {
 	if v == nil {
-		return BucketState{}, nil
+		return []BucketState{}, nil
 	}
 
-	bucketState, ok := v.(BucketState)
+	bucketsState, ok := v.([]BucketState)
 	if !ok {
-		return BucketState{}, microerror.Maskf(wrongTypeError, "expected '%T', got '%T'", BucketState{}, v)
+		return []BucketState{}, microerror.Maskf(wrongTypeError, "expected '%T', got '%T'", []BucketState{}, v)
 	}
 
-	return bucketState, nil
+	return bucketsState, nil
+}
+
+func containsBucketState(bucketState BucketState, bucketStateList []BucketState) bool {
```

Comment:
Why not passing here just the `bucketState` name as a string? Also, this could be unit tested.